tests/extension/oneapi_device_global/device_global_queue_copy.cpp

```diff
@@ -162,36 +162,37 @@ void run_test_copy_from_device_global(util::logger& log,
   // in single_task that will take a significant amount of time
   constexpr size_t numEvents = 5;
   constexpr size_t gen_buf_size = 1000;
-  std::array<usm_api::event_generator<element_type, gen_buf_size>, numEvents>
-      gens{};
+  usm_api::event_generator<element_type, gen_buf_size> gen{};
   element_type init_value{1};
-
-  // List of events that tested usm operation should wait
-  std::vector<sycl::event> depEvents(numEvents);
-  // Run time-consuming tasks to generate events
-  for (size_t i = 0; i < depEvents.size(); ++i) {
-    depEvents[i] = gens[i].init(queue, init_value);
-  }
+  sycl::event depEvent = gen.init(queue, init_value);
 
   // Call overloads with events and check if they wait for events to pass.
   // Use fast copy arrays for future check. If overloads do not
   // wait for the passed events to complete, then an incompletely
   // initialized 'arr_src' will be copied to 'arr_dst' and the test
   // will fail with generator function check().
 
-  auto event2 =
-      queue.copy(dev_global<T>, dst_data2,
-                      sizeof(T) / sizeof(element_type), 0, depEvents[0]);
+  auto event2 = queue.copy(dev_global<T>, dst_data2,
+                           sizeof(T) / sizeof(element_type), 0, depEvent);
   event2.wait();
-  gens[0].copy_arrays(queue);
+  gen.copy_arrays(queue);
+
+  // List of events that tested usm operation should wait
+  std::array<usm_api::event_generator<element_type, gen_buf_size>, numEvents>
+      gens{};
+  std::vector<sycl::event> depEvents(numEvents);
+  // Run time-consuming tasks to generate events
+  for (size_t i = 0; i < depEvents.size(); ++i) {
+    depEvents[i] = gens[i].init(queue, init_value);
+  }
 
   auto event3 = queue.copy(dev_global<T>, dst_data3,
                                 sizeof(T) / sizeof(element_type), 0, depEvents);
   event3.wait();
   // Reverse order is used to increase probability of data race.
   // Array in gens[0] is already copied after event2,
   // so it's not copied here to not to rewrite result for previous case.
-  for (size_t i = numEvents - 1; i > 1; --i) {
+  for (size_t i = numEvents - 1; i + 1 > 0; --i) {
```

Comment:
`i` type is unsigned (size_t is unsigned right?) i.e. `i >= 0`, so `i + 1` is always greater than 0. Isn't this an infinite loop now?

```suggestion
  for (size_t i = numEvents - 1; i - 1 > 0; --i) {
```
?

Reply:
Good catch! That's an oopsie on my part. I have changed it to reverse the range `[numEvents..1]` and instead subtract 1 in the subscript operator. Additionally I have removed the comment about the copy above as it is no longer valid.

Reply:
Thanks. There were two such bugs in this file. You fixed the first one but the second one is still there. Could you update the second one too? Can we refactor the code to avoid code/bugs duplication?

Reply:
Ah, thanks for pointing this out. I have fixed the other issue as well. It may be possible to create a common function for this, but the copy operation uses device global as source in one and dst in the other, so it would be heavily templated and I am not sure it is worth it. If you think it is worth it, I will go refactor it.